### DISCUSSION TITLE
OpenFAST and modelling real world turbines

### DISCUSSION THREAD

**Discussion by clbanjai** [OpenFAST/openfast#2480] [Q&A]
Hello,

I am starting a project where I'll have to simulate some real world wind turbines and was wondering if I could get some guidance from you guys. So far, I've mostly used OpenFAST to simulate the IEA-15-MW-WT using the given fst file. I've been looking into offshore wind turbines such as the ones produced by Siemens and GE, and there's not a lot of information regarding their power curves and CT curves. I found a way to manually estimate the power curve, but the CT curve is more complicated so I was wondering if you knew of a way to generate CT Curves of a turbine given the geometry, rated power, etc using OpenFAST and also overall run simulation for turbines outside of the examples fst files.

Best,

clbanjai

**Comment by jjonkman** [comment]
Dear @clbanjai,

I'm not sure I fully understand your question.  If you have an OpenFAST model of a specific wind turbine, you can use the model to calculate Cp, Ct, etc.

If you don't already have an OpenFAST model of the specific turbine of interest--or the aerodynamic, structural dynamic, and control information needed to generate an OpenFAST model--NREL has been developing capability, based on our WEIS optimization tool (https://github.com/WISDEM/WEIS) that can generate an OpenFAST model based on a limited set of information such as power curve, component masses, and component natural frequencies.  Basically, this capability will start with a reference model (such as the IEA Wind 15-MW RWT) and known information about the specific turbine (power curve, etc.), and reoptimize the turbine so that it has the same properties as the known information, filling in other details (based on assumptions) along the way, with the more information that is known, the better the OpenFAST model will be.

Best regards,

**Reply by clbanjai** [reply]
oh thanks that is very useful, I will be taking a look at WEIS. Additionally, do you happen to know the name of the variable OutListParameters that would correspond to the Ct?

**Reply by jjonkman** [reply]
AeroDyn output `RtAeroCt` is the aerodynamic thrust coefficient of the rotor.

Best regards,

**Comment by mohsamaz** [comment]
Hello everyone,
I am using wisdem. and I have some questons.
in example 10_ccblade(https://wisdem.readthedocs.io/en/master/examples/10_ccblade/tutorial.html#) ,  r shows the center point of each section? or start of each section?
r = np.array(
    [
        2.8667,
        5.6000,
        8.3333,
        11.7500,
        15.8500,
        19.9500,
        24.0500,
        28.1500,
        32.2500,
        36.3500,
        40.4500,
        44.5500,
        48.6500,
        52.7500,
        56.1667,
        58.9000,
        61.6333,
    ]
is there any geometry picture that shows what is exactly Rhub, Rtip,r?

another question is 
why "Cylinder" repeated in af_names? 
af_names = ["Cylinder", "Cylinder", "DU40_A17", "DU35_A17", "DU30_A17", "DU25_A17", "DU21_A17", "NACA64_A17"]

af_idx = [0, 0, 1, 2, 3, 3, 4, 5, 5, 6, 6, 7, 7, 7, 7, 7, 7]

if we can in af_idx  by 0 show 3 times Cylinder section that has been used, but in yaml file in airfoils section there is just one Cylinder? why "Cylinder" repeated? 
Best regards,

**Reply by jjonkman** [reply]
This sounds like a question better posted in the CCBlade repository: https://github.com/WISDEM/CCBlade/issues.
